File: android/app/src/main/java/com/gnomon/agent/core/EnforcementSeam.kt
package com.gnomon.agent.core

/** Version 1 seam only. The production implementation intentionally does nothing. */
interface EnforcementController {
    fun onCategoryExhausted(category: String)
    fun onLockdown(state: Boolean)
}

object NoOpEnforcementController : EnforcementController {
    override fun onCategoryExhausted(category: String) = Unit
    override fun onLockdown(state: Boolean) = Unit
}

File: windows/src/Gnomon.Core/EnforcementSeam.cs
namespace Gnomon.Core;

/// <summary>Version 0.1 seam only. No implementation may enforce policy.</summary>
public interface IEnforcementAdapter
{
    void OnCategoryExhausted(string category);
    void OnLockdown(bool state);
}

public sealed class NoOpEnforcementAdapter : IEnforcementAdapter
{
    public void OnCategoryExhausted(string category) { }
    public void OnLockdown(bool state) { }
}
